### PR TITLE
Add MTP-only training mode

### PR DESCRIPTION
**Pull request by YanivDorGalron**
Freeze backbone parameters and skip their autograd graph so MTP layers train with reduced optimizer and activation memory.

## Description

Add `--freeze-base-model-for-mtp` to train only Multi-Token Prediction (MTP) layers while keeping the base model frozen.

When enabled, this change:

- keeps only `mtp.layers.*` parameters trainable
- excludes backbone parameters from optimizer and DDP gradient buffers
- runs the GPT or Hybrid backbone decoder under `torch.no_grad()` during training
- freezes backbone MoE router expert-bias updates while preserving MTP router updates
- requires `--mtp-num-layers` and rejects use with `--freeze-all-layers`

This reduces optimizer state, gradient storage, and backbone activation memory during MTP-only training.

## Testing

Ran targeted tests on one GPU using `torch.distributed.run`:

- argument validation
- Hybrid decoder gradient context
- MTP forward/backward with and without full recomputation
- verification that only MTP parameters receive gradients and optimizer state

### CODE REVIEW

**Review by janEbert — APPROVED** [review]

**Review by zhongbozhu — APPROVED** [review]

**Review by guihong-nv — APPROVED** [review]

**Review by Phlip79 — APPROVED** [review]

**Review by jaredcasper — APPROVED** [review]

**Review by fanshiqing — APPROVED** [review]

File: megatron/training/training.py [hunk 1]
```diff
@@ -2365,6 +2365,48 @@ def _freeze_all_model_chunks(model_list):
...
+
+    print_rank_0(
+        f'[freeze-base-model-for-mtp] Frozen {frozen_params:,} backbone parameters and '
+        f'{frozen_router_biases:,} backbone router expert-bias buffers. '
+        f'Trainable MTP parameters: {trainable_params:,}.'
+    )
+    return model_list
+
+
+def _add_model_freeze_pre_wrap_hook(model_config, args):
```
Comment (maanug-nv) [inline megatron/training/training.py]: one nit. since only 2 settings from args are needed here, can we pass those explicitly? imo limiting state that's exposed in these small funcs keeps them focused, avoids vaguely related logic from being added just because args is available.

**Review by maanug-nv — APPROVED** [review]

**Comment by cuichenx** [comment]
/ok to test https://github.com/NVIDIA/Megatron-LM/pull/7021/commits/603059ea36f566cd437fe85099fad6be85944d10
